mkosi/run.py

```diff
@@ -304,9 +306,21 @@ def run(
     except FileNotFoundError as e:
         die(f"{e.filename} not found.")
     except subprocess.CalledProcessError as e:
+        if ARG_DEBUG_SHELL.get():
+            subprocess.run(
+                [*sandbox, "sh"],
+                check=False,
+                stdin=sys.stdin,
+                text=True,
+                user=user,
+                group=group,
+                env=env,
+                cwd=cwd,
+                preexec_fn=make_foreground_process,
+            )
         if log:
             log_process_failure(cmdline, e.returncode)
-        raise e
+        raise subprocess.CalledProcessError(e.returncode, cmdline, output=e.output, stderr=e.stderr)
```

Comment:
What's the reason for constructing a new error here?

Reply:
To get rid of the sandboxing from the command line so we don't show it to the user by default.

Reply:
Makes sense. We could construct a new horror here and also add a `from e`, this way one would see both stack traces and could scroll up to see the old context. Might be handy for debugging.

Reply:
I opted to just change `e.cmd` instead of doing the chaining. I think that leads to a more readable stacktrace and if users want the full command they can run with `--debug`